### ISSUE TITLE
Perfectly wrong and use one responder per action

### ISSUE THREAD

**Issue by harikt** [harikt/radar-experiments#1]
Need to move this piece : https://github.com/harikt/radar-experiments/blob/aed586c78e6fb1a695cc78f88e6b579d0f1d21a5/src/HtmlResponder.php#L58-L59

Responder per Action is the real candidate.

**Comment by cxj** [comment]
Yes, using `Aura\Payload\Payload::setExtras()` to pass the template name to the Responder seems wrong.  `Payload` is what is returned by the Domain layer, and the Domain layer should not know anything about HTML templates (presentation).

While having one Responder per Action would solve this problem, it requires an application with many Actions to have a lot more code.  Most of the time, templates are going to be constructed so as to be as generic as possible.  That is, many templates will be re-used by many Responders.  It seems like it would be nicer to find another way to pass the template name to the Responder.

Would it be appropriate to use `Aura\Router\Route::extras()` or `Aura\Router\Route::defaults()` via their `Radar\Adr\Adr` accessors to do this, do you think?

**Comment by harikt** [comment]
@cxj I don't really know, but this was recently came into my mind.

What about having an event dispatcher sending the information. And before respond the event listener can catch the event and render making use of content negotiation. I am still thinking about it though. I am looking at https://github.com/akrabat/rka-content-type-renderer

**Comment by cxj** [comment]
I have no need in my projects for content negotiation, but that's an interesting project.  It adds complexity, so I probably won't use it.

Your code works perfectly as is, but it causes some coupling that should not be there.  I will think about it more tomorrow and see if I can find a solution.

**Comment by harikt** [comment]
@cxj sure :) .

Another idea is keep all the names in the responder and set controller name instead of view ?

```php
$views = [
    WelcomeController::class => 'welcome',
    SomeController::class => 'some',
    SomeController::class => 'some',
];
```

The benefit is you can change the view names in one place.
